pallets/resource-discussions/src/lib.rs

```diff
@@ -0,0 +1,132 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+// pub use crate::weights::WeightInfo;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+// #[cfg(feature = "runtime-benchmarks")]
+// mod benchmarking;
+// pub mod weights;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use frame_support::{dispatch::RawOrigin, pallet_prelude::*, transactional};
+    use frame_system::pallet_prelude::*;
+    use sp_std::convert::TryInto;
+
+    use pallet_posts::{NextPostId, PostExtension};
+    use subsocial_support::{Content, PostId};
+
+    // use crate::weights::WeightInfo;
+
+    type ResourceId<T> = BoundedVec<u8, <T as Config>::MaxResourceIdLength>;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config + pallet_posts::Config + pallet_spaces::Config {
+        /// The overarching event type.
+        type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+        /// The maximum number of characters in a resource id.
+        #[pallet::constant]
+        type MaxResourceIdLength: Get<u32>;
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        ResourceDiscussionLinked {
```

Comment:
Think about a better name for this event. It is used in both situations: 
- When we are creating a new resource discussion.
- When we are linking a new post to the resource.

Maybe it should be called `ResourceDiscussionUpdated`?